code/game/objects/items/ego_weapons/waw.dm

```diff
@@ -15,15 +17,27 @@
 							PRUDENCE_ATTRIBUTE = 60
 							)
 
+/obj/item/ego_weapon/lamp/attack(mob/living/M, mob/living/user)
+	..()
+	for(var/mob/living/L in livinginrange(1, M))
+		force = 25
+		var/userjust = (get_attribute_level(user, JUSTICE_ATTRIBUTE))
+		var/justicemod = 1 + userjust/100
+		force*=justicemod
+		if(L == user)
```

Comment:
You never check for faction, meaning that this weapon will kill everyone nearby. This is VERY bad for a melee weapon.

Reply:
I planned not to, but I can change it to do so.